front/pages/api/login.ts

```diff
@@ -310,27 +307,24 @@ async function handler(
   }
 }
 
-async function _createAndLogMembership(data: {
+async function _createAndLogMembership({
+  userId,
+  workspace,
+  role,
+}: {
   userId: number;
-  workspaceId: number;
-  workspaceName: string;
+  workspace: Workspace;
   role: "admin" | "user";
-  invitationFlow: "domain" | "email" | "personal";
 }) {
   const m = await Membership.create({
-    role: data.role,
-    userId: data.userId,
-    workspaceId: data.workspaceId,
+    role: role,
+    userId: userId,
+    workspaceId: workspace.id,
   });
 
-  // const tags = [
-  //   `workspace_id:${data.workspaceId}`,
-  //   `workspace_name:${data.workspaceName}`,
-  //   `user_id:${data.userId}`,
-  //   `role:${data.role}`,
-  //   `invitation_flow:${data.invitationFlow}`,
-  // ];
-  // statsDClient.increment("workspace.membership_created", 1, tags);
+  // If the user is joining a workspace with a subscription based on monthly_active_seats,
+  // we need to update the Stripe subscription quantity.
+  await updateWorkspacePerSeatSubscriptionUsage({ workspaceId: workspace.sId });
```

Comment:
This function can throw based on the rest of the code in this PR. We def don't want it to throw and I think we definitly don't want to await it?

Reply:
I wrapped the try/catch on the whole function, and removed the await in there as indeed we don't want to await it.

Reply:
Cool re try/catch let's add a comment here to say that it's safe (no error thrown) and void call it :+1: